src/main/java/org/mozilla/jss/JSSProvider.java

```diff
@@ -88,6 +104,34 @@ public CryptoManager getCryptoManager() {
         return cm;
     }
 
+    @Override
+    public Service getService(String type, String algorithm) {
+        if (isAlgorithmDeprecated(algorithm)) {
+            Optional<StackFrame> stackFrame = StackWalker.getInstance().walk(s -> s.skip(5).findFirst());
```

Comment:
This should work, but in case OpenJDK changes the implementation the number of frames to skip might have to be adjusted later.  Another alternative is to find the `MessageDigest.getInstance()` call in the stack, then go to the next frame to find the caller.

Reply:
It is already broken, there are multiple paths of calls that can be taken depending on `Provider` with different numbers of levels so I have re-implemented to check for `MessageDigest.getInstance()`. It's not as pretty as the `Stream` API, but it should be more robust.